client/state/selectors/is-eligible-for-checkout-to-checklist.js

```diff
@@ -10,26 +10,31 @@ import {
 	hasTransferProduct,
 	hasPlan,
 	hasConciergeSession,
-	hasEcommercePlan,
 } from 'lib/cart-values/cart-items';
 import isEligibleForDotcomChecklist from './is-eligible-for-dotcom-checklist';
+import { retrieveSignupDestination } from 'signup/utils';
 
 /**
  * @param {Object} state Global state tree
  * @param {Number} siteId Site ID
  * @param {Object} cart object
+ * @param {String} destination The final destination URL after signup completes
  * @return {Boolean} True if current user is able to see the checklist after checkout
  */
-export default function isEligibleForCheckoutToChecklist( state, siteId, cart ) {
+export default function isEligibleForSignupDestination( state, siteId, cart ) {
 	if (
 		hasDomainMapping( cart ) ||
 		hasDomainRegistration( cart ) ||
 		hasTransferProduct( cart ) ||
-		( ! hasPlan( cart ) && ! hasConciergeSession( cart ) ) ||
-		hasEcommercePlan( cart )
+		( ! hasPlan( cart ) && ! hasConciergeSession( cart ) )
 	) {
 		return false;
 	}
 
-	return isNewSite( state, siteId ) && isEligibleForDotcomChecklist( state, siteId );
+	const destination = retrieveSignupDestination();
+	if ( destination.includes( '/checklist/' ) ) {
+		return isNewSite( state, siteId ) && isEligibleForDotcomChecklist( state, siteId );
```

Comment:
We were earlier running the `isEligibleForDotcomChecklist ` check even when the destination when was not checklist. It makes sense to restrict this check only for /checklist destination routes.